test_runner/regress/test_lsn_mapping.py

```diff
@@ -108,6 +111,64 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
 
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    """
+    Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+    Added as an effort to improve error handling and avoid full anyhow backtrace.
+    """
+
+    env = neon_env_builder.init_start()
+    env.pageserver.allowed_errors.extend(
+        [
+            ".*request was dropped before completing.*",
+            ".*Cancelled request finished with an error: Cancelled\n",
```

Comment:
This newline at the end of this line is curious.

The way python re matching works is that it matches a prefix of the string, which is a weird "match" (usually matching means that the whole string needs to match), but I am surprised that we'd need this `\n` as all lines end with that. Perhaps I would had expected `$` instead of `\n` (or neither).